tracing/src/main/java/com/palantir/remoting2/tracing/Tracer.java

```diff
@@ -114,6 +114,17 @@ private static OpenSpan startSpanInternal(String operation, SpanType type) {
      * completed span.
      */
     public static Optional<Span> completeSpan() {
+        return completeSpanInternal(Collections.emptyMap());
+    }
+
+    /**
+     * Like {@link #completeSpan()}, but adds {@code metadata} to the current span being completed.
+     */
+    public static Optional<Span> completeSpanWithMetadata(Map<String, String> metadata) {
```

Comment:
i don't feel very strongly about it, do you see any specific advantages? `SpanMetadata` provides some additional readability but a `metadata` member on the `Span` class doesn't feel too hard to understand either. It could help future proof us wanting to handle other binary annotation Types but that may be premature.